proposals/text/7462-rag-evaluation.md

```diff
@@ -0,0 +1,243 @@
+- Title: Proposal for presentation of RAG evaluation results
+- Decision driver: David S. Batista
+- Start Date: 2024-04-03
+- Proposal PR: #7462
+- Github Issue or Discussion: https://github.com/deepset-ai/haystack/issues/7398
+
+# Summary
+
+Add a new component to Haystack allowing users interact with the results of evaluating the performance of a RAG model.
+
+
+# Motivation
+
+RAG models are one of them most popular use cases for Haystack. We are adding support for evaluations metrics, but there is no way to present the results of the evaluation.
+
+
+# Detailed design
+
+The output results of an evaluation pipeline composed of `evaluator` components are passed to a `RAGPipelineEvaluation`
+(this is a placeholder name) which stores them internally and acts as an interface to access and present the results.
+
+Example of the data structure that the `RAGPipelineEvaluation` class will receive for initialization:
+
+```python
+
+data = {
+    "queries": {
+        "query_id": ["53c3b3e6", "225f87f7"],
+        "question": ["What is the capital of France?", "What is the capital of Spain?"],
+        "contexts": ["wiki_France", "wiki_Spain"],
+        "answer": ["Paris", "Madrid"]
+    },
+    "metrics":
+        [
+            {"name": "reciprocal_rank", "scores": [0.378064, 0.534964, 0.216058, 0.778642]},
+            {"name": "single_hit", "scores": [1, 1, 0, 1]},
+            {"name": "multi_hit", "scores": [0.706125, 0.454976, 0.445512, 0.250522]},
+            {"name": "context_relevance", "scores": [0.805466, 0.410251, 0.750070, 0.361332]},
+            {"name": "faithfulness", "scores": [0.135581, 0.695974, 0.749861, 0.041999]},
+            {"name": "semantic_answer_similarity", "scores": [0.971241, 0.159320, 0.019722, 1]}
+         ],
+    "pipeline_answers": ["Paris", "Madrid"]
+    },
+
+```
+
+- At least the `query_id` or the `question and context` should be present in the data structure.
+- At least one of the metrics should be present in the data structure.
+- The `pipeline_answers` field is optional, it is used to compare the answers generated by the pipeline with the expected answers.
+
+
+The `RAGPipelineEvaluation` class provides the following methods to different types of users:
+
+Basic users:
+- `evaluation_report()`
```

Comment:
Let's either pick _summary_ or _report_ and stick with it for consistency. I personally prefer _report_.

Also, let's be explicit in our naming - something along the lines of `aggregate_score_report` is more informative than `evaluation_report`.